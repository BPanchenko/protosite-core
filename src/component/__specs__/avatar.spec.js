import AvatarComponent from '../avatar.js'
import compactObject from '../../lib/cb.compactObject.js'

describe('[AvatarComponent]', () => {
	describe('Attributes:', () => {
		describe.each`
			img                                            | href                   | target
			${undefined}                                   | ${undefined}           | ${'_top'}
			${'http://protosite.rocks/pictures/ava.gif'}   | ${'http://yandex.ru'}  | ${undefined}
			${'http://protosite.rocks/pictures/ava-1.gif'} | ${undefined}           | ${'_parent'}
			${undefined}                                   | ${'http://google.com'} | ${'_blank'}
		`('[Image = "$img"; Link = "$href"; Target = "$target"]', (options) => {
			const avatar = new AvatarComponent(compactObject(options))

			beforeEach(() => document.body.appendChild(avatar))
			afterEach(() => avatar.remove())

			it('Light DOM:', () => {
				expect(avatar).toMatchSnapshot()
			})
			it('Shadow DOM:', () => {
				expect(avatar.shadowRoot.children).toMatchSnapshot()
			})
		})

		describe.each`
			size
			${undefined}
			${'xxs'}
			${'xs'}
			${'sm'}
			${'md'}
			${'lg'}
			${'xl'}
			${'xxl'}
		`('[Size = "$size"]', (options) => {
			const avatar = new AvatarComponent(compactObject(options))

			beforeEach(() => document.body.appendChild(avatar))
			afterEach(() => avatar.remove())

			it('Light DOM:', () => {
				expect(avatar).toMatchSnapshot()
			})
			it('Shadow DOM:', () => {
				expect(avatar.shadowRoot.children).toMatchSnapshot()
			})
		})
	})
})
