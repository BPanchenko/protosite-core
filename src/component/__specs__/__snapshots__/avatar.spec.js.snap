// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[AvatarComponent] Attributes: [Image = "http://protosite.rocks/pictures/ava.gif"; Link = "http://yandex.ru"; Target = "undefined"] Light DOM: 1`] = `
<c-avatar
  href="http://yandex.ru"
  img="http://protosite.rocks/pictures/ava.gif"
>
  <a
    class="c-avatar__link"
    href="http://yandex.ru"
    target="_self"
  >
    <img
      src="http://protosite.rocks/pictures/ava.gif"
    />
  </a>
</c-avatar>
`;

exports[`[AvatarComponent] Attributes: [Image = "http://protosite.rocks/pictures/ava.gif"; Link = "http://yandex.ru"; Target = "undefined"] Shadow DOM: 1`] = `
HTMLCollection [
  <div
    role="img"
  >
    <slot />
  </div>,
]
`;

exports[`[AvatarComponent] Attributes: [Image = "http://protosite.rocks/pictures/ava-1.gif"; Link = "undefined"; Target = "_parent"] Light DOM: 1`] = `
<c-avatar
  img="http://protosite.rocks/pictures/ava-1.gif"
  target="_parent"
>
  <img
    src="http://protosite.rocks/pictures/ava-1.gif"
  />
</c-avatar>
`;

exports[`[AvatarComponent] Attributes: [Image = "http://protosite.rocks/pictures/ava-1.gif"; Link = "undefined"; Target = "_parent"] Shadow DOM: 1`] = `
HTMLCollection [
  <div
    role="img"
  >
    <slot />
  </div>,
]
`;

exports[`[AvatarComponent] Attributes: [Image = "undefined"; Link = "http://google.com"; Target = "_blank"] Light DOM: 1`] = `
<c-avatar
  href="http://google.com"
  target="_blank"
>
  <a
    class="c-avatar__link"
    href="http://google.com"
    target="_blank"
  />
</c-avatar>
`;

exports[`[AvatarComponent] Attributes: [Image = "undefined"; Link = "http://google.com"; Target = "_blank"] Shadow DOM: 1`] = `
HTMLCollection [
  <div
    role="img"
  >
    <slot />
  </div>,
]
`;

exports[`[AvatarComponent] Attributes: [Image = "undefined"; Link = "undefined"; Target = "_top"] Light DOM: 1`] = `
<c-avatar
  target="_top"
/>
`;

exports[`[AvatarComponent] Attributes: [Image = "undefined"; Link = "undefined"; Target = "_top"] Shadow DOM: 1`] = `
HTMLCollection [
  <div
    role="img"
  >
    <slot />
  </div>,
]
`;

exports[`[AvatarComponent] Attributes: [Size = "lg"] Light DOM: 1`] = `
<c-avatar
  size="lg"
/>
`;

exports[`[AvatarComponent] Attributes: [Size = "lg"] Shadow DOM: 1`] = `
HTMLCollection [
  <div
    class="s-lg"
    role="img"
  >
    <slot />
  </div>,
]
`;

exports[`[AvatarComponent] Attributes: [Size = "md"] Light DOM: 1`] = `
<c-avatar
  size="md"
/>
`;

exports[`[AvatarComponent] Attributes: [Size = "md"] Shadow DOM: 1`] = `
HTMLCollection [
  <div
    class="s-md"
    role="img"
  >
    <slot />
  </div>,
]
`;

exports[`[AvatarComponent] Attributes: [Size = "sm"] Light DOM: 1`] = `
<c-avatar
  size="sm"
/>
`;

exports[`[AvatarComponent] Attributes: [Size = "sm"] Shadow DOM: 1`] = `
HTMLCollection [
  <div
    class="s-sm"
    role="img"
  >
    <slot />
  </div>,
]
`;

exports[`[AvatarComponent] Attributes: [Size = "undefined"] Light DOM: 1`] = `<c-avatar />`;

exports[`[AvatarComponent] Attributes: [Size = "undefined"] Shadow DOM: 1`] = `
HTMLCollection [
  <div
    role="img"
  >
    <slot />
  </div>,
]
`;

exports[`[AvatarComponent] Attributes: [Size = "xl"] Light DOM: 1`] = `
<c-avatar
  size="xl"
/>
`;

exports[`[AvatarComponent] Attributes: [Size = "xl"] Shadow DOM: 1`] = `
HTMLCollection [
  <div
    class="s-xl"
    role="img"
  >
    <slot />
  </div>,
]
`;

exports[`[AvatarComponent] Attributes: [Size = "xs"] Light DOM: 1`] = `
<c-avatar
  size="xs"
/>
`;

exports[`[AvatarComponent] Attributes: [Size = "xs"] Shadow DOM: 1`] = `
HTMLCollection [
  <div
    class="s-xs"
    role="img"
  >
    <slot />
  </div>,
]
`;

exports[`[AvatarComponent] Attributes: [Size = "xxl"] Light DOM: 1`] = `
<c-avatar
  size="xxl"
/>
`;

exports[`[AvatarComponent] Attributes: [Size = "xxl"] Shadow DOM: 1`] = `
HTMLCollection [
  <div
    class="s-xxl"
    role="img"
  >
    <slot />
  </div>,
]
`;

exports[`[AvatarComponent] Attributes: [Size = "xxs"] Light DOM: 1`] = `
<c-avatar
  size="xxs"
/>
`;

exports[`[AvatarComponent] Attributes: [Size = "xxs"] Shadow DOM: 1`] = `
HTMLCollection [
  <div
    class="s-xxs"
    role="img"
  >
    <slot />
  </div>,
]
`;
